wts/resources/oauth2.py

```diff
@@ -84,9 +85,14 @@ def refresh_refresh_token(tokens, idp):
     user = current_user
     username = user.username
 
+    if "keycloak" in idp:
+        idp_username = id_token["email"]
```

Comment:
similar to the above comment, would prefer not to have this keycloak string hardcoded here. I'm not sure I like the idea of having to extend this if condition for all new IdPs. It's probably okay since the code here is small. Unless you can think of a better way to abstract all this.

Reply:
i'll suggest a config like `id_token_username_field` that you can set to `email` for keycloak and that defaults to `context.user.name` for fence integrations

Reply:
Agreed. You should check out how Fence handles the abstractions of dealing with different IdPs and maybe try to mirror those abstractions here.

Reply:
Done!